### ISSUE TITLE
UX improvement: use XDG Base Directory Specification

### ISSUE THREAD

**Issue by ceciltilley898** [yairm210/Unciv#10710]
### Before creating

- [X] This is NOT a gameplay feature from Civ VI, BNW, or outside - see Roadmap
- [X] This is NOT a gameplay feature from Vanilla Civ V or from G&K - If so, it should be a comment in https://github.com/yairm210/Unciv/issues/4697

### Problem Description

By default, Unciv stores "GameSettings.json" file and "SaveFiles" directory in current working directory

### Desired Solution

By default, Unciv should use [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) (also on [ArchWiki](https://wiki.archlinux.org/title/XDG_Base_Directory)) to store "GameSettings.json" file in "${XDG_CONFIG_HOME}/Unciv" directory and "SaveFiles" directory in "${XDG_DATA_HOME}/Unciv" directory without the user having to set it up after the first time they start the game. Both directories should be automatically created by the game if missing

### Alternative Approaches

"SaveFiles" directory could be ommited so game save files could be stored directly in "${XDG_DATA_HOME}/Unciv" directory

**Comment by yairm210** [comment]
This sounds like input parameters to the .jar file, that whatever runs Unciv on Arch will supply via command line.
What *does* run Unciv on Arch?
https://aur.archlinux.org/packages/unciv-bin maintained by @Thyrum 
So this is a joint venture

My only worry is that this will break existing games by changing the directories under them, not sure how to deal with that

**Comment by Thyrum** [comment]
> This sounds like input parameters to the .jar file, that whatever runs Unciv on Arch will supply via command line. What _does_ run Unciv on Arch? https://aur.archlinux.org/packages/unciv-bin maintained by @Thyrum So this is a joint venture

For Arch Linux, all installation information is stored in the [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=unciv-bin). Under "source" it first downloads and extracts the `linuxFilesForJar.zip` relevant for this release and under "package" it installs `Unciv.sh` as `$pkgdir/usr/bin/Unciv` (so all parameters for the `.jar` must be specified in `Unciv.sh`). Note that for many default installations `$pkgdir` is the root directory, though it can be changed for a custom installpath. This would break the current setup though, as `Unciv.sh` doesn't account for this and just calls `java -jar /usr/share/Unciv/Unciv.jar`. I should probably fix this at some point.

TL;DR: For the Arch Linux installation, `Unciv` is run using the `Unciv.sh` file contained in the `linuxFilesForJar.zip` from the current release.

**Comment by Thyrum** [comment]
> But what is the cwd the moment the DesktopLauncher gets control? ...

The `Unciv.sh` script changes directory to `CONFIG_DIR="$HOME/.local/share/Unciv` before starting Java. It also has an optional `--config-dir` parameter which can be used to execute with a different `CONFIG_DIR`.

So if I am understanding this correctly, the `Unciv.sh` script will indeed behave as we want when adding support for `XDG_DATA_HOME`, though I would also recommend editing the `Unciv.sh` script so that it also uses `XDG_DATA_HOME` as the default.

**Comment by yairm210** [comment]
The Unciv side has been solved - we now accept a "--data-dir=<your_location>" argument when starting the JAR
I'll need to see if the 'packed" executables transfer arguments *to* the JAR, but I assume that they do

**Comment by yairm210** [comment]
Yes indeed it works, which means that whether you're running through java -jar or through the "minified jre" you are now able to specify the data directory as an argument :)
